src/lib.rs

```diff
@@ -385,17 +333,17 @@ impl<M: Manager> Pool<M> {
         });
 
         let pool_state = PoolState {
-            num_open: AtomicU64::new(0),
+            num_open: Arc::new(AtomicU64::new(0)),
             max_lifetime_closed: AtomicU64::new(0),
             wait_count: AtomicU64::new(0),
-            max_idle_closed: AtomicU64::new(0),
+            max_idle_closed: Arc::new(AtomicU64::new(0)),
         };
 
         let shared = Arc::new(SharedPool {
             config: share_config,
             manager,
             internals,
-            semaphore: Semaphore::new(max_open),
+            semaphore: Arc::new(Semaphore::new(max_open)),
```

Comment:
Why do you need the Arc around the Semaphore? 
What is the performance implications of this?

Reply:
`Arc` is needed to get owned permit, that method exists only on `Arc<Self>`
No noticeable negative performance implications, if anything, mobc own testsuite consistently runs 10-15s faster on my machine after my changes.